polyfill/lib/ecmascript.mjs

```diff
@@ -3467,11 +3446,19 @@ export const ES = ObjectAssign({}, ES2020, {
           days -= oneYearDays;
           ({ relativeTo, days: oneYearDays } = ES.MoveRelativeDate(calendar, relativeTo, oneYear));
         }
-        years += days / Math.abs(oneYearDays);
-
-        total = years;
-        years = ES.RoundNumberToIncrement(years, increment, roundingMode);
-        months = weeks = days = 0;
+        // Note that `nanoseconds` below (here and in similar code for months,
+        // weeks, and days further below) isn't actually nanoseconds for the
+        // full date range.  Instead, it's a BigInt representation of total
+        // days multiplied by the number of nanoseconds in the last day of
+        // the duration. This lets us do days-or-larger rounding using BigInt
+        // math which reduces precision loss.
+        oneYearDays = Math.abs(oneYearDays);
```

Comment:
Looks like we need a full sweep at some point.

Reply:
Hi @Ms2ger - I went through and switched all usage in ecmascript.js of `Math.*` and replaced them with aliases like `MathAbs`. I'm assuming that this guidance extends to other built-ins too, so while I was at it I added aliases for `Number.*` and `Object.*` static methods too.

Legacy Date is used (constructor, static now(), and prototype methods) but I wasn't sure if this is intentional (for SES) or not. So I didn't make any changes to its usage.

There were built-in prototype methods used too, but I wasn't sure if those needed to be replaced too so didn't change any. Here's the prototype methods I found that were used: 
* Regex.prototype.exec
* String.prototype.slice
* String.prototype.match
* String.prototype.split
* Array.prototype.map
* Array.prototype.filter
* Array.prototype.includes
* Array.prototype.some
* Array.prototype.push
* Array.prototype.unshift
* Array.prototype.join
* Set (constructor)
* Set.prorotype.has
* Set.prototype.delete
* Map (constructor)
* Map.prototype.has
* Map.prototype.get
* Map.prototype.keys
* Map.prototype.values

Finally, there's lots of use of prototype methods on `big-integer`, but I assumed because as an imported library there's no need to change those calls. LMK if this assumption was wrong.

To avoid holding up this PR, I'll open a separate issue for `Date` and prototype methods.

Reply:
I filed #1222 to track the non-aliased use of built-ins per the comment above. Also, I noticed that there's some use of Math.*, Object.*, etc. static methods outside ecmascript.js. I didn't make any changes to those files since it seemed out of scope to this PR, but I noted this in #1222 for later attention.